pkgs/applications/graphics/mcomix3/default.nix

```diff
@@ -0,0 +1,90 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, python3
+, python3Packages
+, wrapGAppsHook
+, gobject-introspection
+, gtk3
+, gdk-pixbuf
+# Recommended Dependencies:
+, unrarSupport ? false
+, unrar
+, p7zip
+, lhasa
+, mupdf
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "mcomix3";
+  version = "20201123";
+
+  # fetch from github because no official release on pypi/github and no build system
+  src = fetchFromGitHub {
+    repo   = "${pname}";
+    owner  = "multiSnow";
+    rev = "cdcb27533dc7ee2ebf7b0a8ab5ba10e61c0b8ff8";
+    sha256 = "0q9xgl60ryf7qmy5vgzgfry4rvw5j9rb4d1ilxmpjmvm7dd3fm2k";
+  };
+
+  buildInputs = [ gobject-introspection p7zip lhasa mupdf ]
+    ++ lib.optional (unrarSupport) unrar;
+  nativeBuildInputs = [ wrapGAppsHook ];
+  propagatedBuildInputs = (with python3Packages; [ pillow pygobject3 pycairo ])
+    ++ [ gtk3 gdk-pixbuf ];
```

Comment:
These two still should not be propagated, only Python dependencies need to be and [soon not even those](https://github.com/NixOS/nixpkgs/pull/102613).

Reply:
Weird, I could've sworn that I had them in `buildInputs` before and things were not working then. That's why I moved them at some point. But now that I moved them again, it's fine. I'm very happy, when this thing is eventually merged, I'm starting to go crazy.